intel_adsp: xt-clang: unknown warning group -Wunknown-warning-option.

**Describe the bug**
intel_adsp/ace15_mtpm/sim
```
INFO    - 1) tests/lib/mem_alloc/libraries.libc.minimal.mem_alloc on intel_adsp/ace15_mtpm/sim error (Build failure)
INFO    - 2) tests/lib/mem_alloc/libraries.libc.minimal.mem_alloc_negative_testing on intel_adsp/ace15_mtpm/sim error (Build failure)
INFO    - 3) tests/lib/mem_alloc/libraries.libc.picolibc.mem_alloc on intel_adsp/ace15_mtpm/sim error (Build failure)
INFO    - 4) tests/kernel/mem_protect/stack_random/kernel.memory_protection.stack_random on intel_adsp/ace15_mtpm/sim error (Build failure)
```

intel_adsp/ace30/ptl/sim
```
INFO    - 1) tests/kernel/mem_protect/stack_random/kernel.memory_protection.stack_random on intel_adsp/ace30/ptl/sim error (Build failure)
INFO    - 2) tests/lib/mem_alloc/libraries.libc.minimal.mem_alloc on intel_adsp/ace30/ptl/sim error (Build failure)
INFO    - 3) tests/lib/mem_alloc/libraries.libc.minimal.mem_alloc_negative_testing on intel_adsp/ace30/ptl/sim error (Build failure)
INFO    - 4) tests/lib/mem_alloc/libraries.libc.picolibc.mem_alloc on intel_adsp/ace30/ptl/sim error (Build failure)
```

intel_adsp/ace20_lnl/sim
```
INFO    - 1) tests/lib/mem_alloc/libraries.libc.minimal.mem_alloc on intel_adsp/ace20_lnl/sim error (Build failure)
INFO    - 2) tests/lib/mem_alloc/libraries.libc.minimal.mem_alloc_negative_testing on intel_adsp/ace20_lnl/sim error (Build failure)
INFO    - 3) tests/lib/mem_alloc/libraries.libc.picolibc.mem_alloc on intel_adsp/ace20_lnl/sim error (Build failure)
INFO    - 4) tests/kernel/mem_protect/stack_random/kernel.memory_protection.stack_random on intel_adsp/ace20_lnl/sim error (Build failure)
```


**To Reproduce**
`twister -p intel_adsp/ace20_lnl/sim -T  tests/lib/mem_alloc/`


**Logs and console output**
`/tests/lib/mem_alloc/src/main.c:24:32: error: unknown warning group '-Walloc-size-larger-than=', ignored [-Werror,-Wunknown-warning-option]
#pragma GCC diagnostic ignored "-Walloc-size-larger-than="`

**Environment:**

 - OS: Linux
 - Toolchain: xt-clang
 - Commit SHA or Version used: v4.0.0-3486-g4cc76e8a9ae3

**Additional context**
b9d7b1bdae9b41d3a73070536821c481f77633f1 is the first bad commit
```
commit b9d7b1bdae9b41d3a73070536821c481f77633f1
Author: Tom Hughes <tomhughes@chromium.org>
Date:   Mon Jan 13 12:16:12 2025 -0800

    cmake clang/compiler_flags.cmake: Re-enable -Wunknown-warning-option

    All warnings in the code base have been resolved.

    Signed-off-by: Tom Hughes <tomhughes@chromium.org>
```

--- a/cmake/compiler/xt-clang/compiler_flags.cmake
+++ b/cmake/compiler/xt-clang/compiler_flags.cmake
@@ -34,3 +34,12 @@ set_compiler_property(PROPERTY no_position_independent "")
 
 # Remove after testing that -Wshadow works
 set_compiler_property(PROPERTY warning_shadow_variables)
+
+# xt-clang is usually based on older version of clang, and
+# Zephyr main targets more recent versions. Because of this,
+# some newer compiler flags may cause warnings where twister
+# would mark as test being failed. To workaround that,
+# add -Wno-unknown-warning-option to suppress those warnings.
+check_set_compiler_property(APPEND PROPERTY warning_extended
+                            -Wno-unknown-warning-option
+)

